### PR TITLE
*: remove unix socket.

### PR DESCRIPTION
**Pull request by disksing**
As etcd does not support unix socket, using unix socket in tests improves nothing and introduces complexity.

cc @nolouch @siddontang @huachaohuang

### CODE REVIEW

File: server/api/server_test.go [hunk 1]
```diff
@@ -50,43 +48,21 @@ var (
...
 		Peers: peers,
 	}
-	unixClient = newUnixSocketClient()
 )
 
 func TestAPIServer(t *testing.T) {
 	TestingT(t)
 }
 
 func newUnixSocketClient() *http.Client {
```
Comment (siddontang) [inline server/api/server_test.go]: need to remove this?

File: pkg/testutil/testutil.go [hunk 2]
```diff
@@ -15,12 +15,20 @@ package testutil
...
-// UnixURL returns a unique unix socket url, used for test only.
-func UnixURL() string {
-	return fmt.Sprintf("unix://localhost:%d", atomic.AddUint64(&unixURLCount, 1))
+// AllocTestURL allocates a local URL for testing.
+func AllocTestURL() string {
+	l, err := net.Listen("tcp", "127.0.0.1:0")
+	if err != nil {
+		log.Fatal(err)
+	}
+	err = l.Close()
```
Comment (siddontang) [inline pkg/testutil/testutil.go]: should get the address before close?

File: server/api/redirector.go [hunk 3]
```diff
@@ -79,18 +79,8 @@ func newCustomReverseProxies(urls []url.URL) *customReverseProxies {
...
-		if u.Scheme == "unix" {
-			u.Scheme = "http"
-			client = &http.Client{Transport: &http.Transport{Dial: unixDial}}
-		} else {
-			client = &http.Client{}
-		}
-
 		p.urls = append(p.urls, u)
-		p.clients = append(p.clients, client)
+		p.clients = append(p.clients, &http.Client{})
```
Comment (huachaohuang) [inline server/api/redirector.go]: We don't need multiple clients anymore.

**Comment by huachaohuang** [comment]
LGTM

**Pull request merged**